src/Compilers/CSharp/Portable/Syntax/TypeDeclarationSyntax.cs

```diff
@@ -115,7 +115,7 @@ public static TypeDeclarationSyntax TypeDeclaration(SyntaxKind kind, string iden
                 case SyntaxKind.RecordDeclaration:
                     return SyntaxFactory.RecordDeclaration(attributes, modifiers, keyword, identifier, typeParameterList, parameterList: null, baseList, constraintClauses, openBraceToken, members, closeBraceToken, semicolonToken);
                 default:
-                    throw new ArgumentException("kind");
+                    throw new ArgumentException(CSharpResources.SyntaxKindIncorrectDeclarationType, nameof(kind));
```

Comment:
does the compiler generally produce tehse specific error messages for SyntaxFactory methods?  If not, this feels liek it would be out of place to have in this single place?  While i'm cognizant of hte fact that the error message helps here, it is a maintenance burden and is inconsistent.

I woudl be more in support of a generic `"Syntax kind must be: {0}` error, and then have teh *syntax generator* emit all these argument calls that tell you exactly what kinds are allowed, instead of htis one-off approach.

Reply:
I agree with the suggested error message output. 
About first question: 
As far as I remember, the method is only called at the point indicated in the [previous comment ](https://github.com/dotnet/roslyn/pull/51212#issuecomment-782353991). I can check it later.

Reply:
Just checked CSharpResources.
This file have too much hardcoded strings like: 
`<value>'{0}' is not a valid constraint. A type used as a constraint must be an interface, a non-sealed class or a type parameter.</value>`
 or 
`<value>'{0}' is of type '{1}'. The type specified in a constant declaration must be sbyte, byte, short, ushort, int, uint, long, ulong, char, float, double, decimal, bool, string, an enum-type, or a reference-type.</value>`
But not found generic error examples. Maybe remake my message to: 
`Syntax kind is {0}. The syntax kind must class, struct, interface or record declaration`?

If not, can you help me with generator example please?